benchmarks/shared/python/data_generators/gnn_cite_graph_gen.py

```diff
@@ -0,0 +1,826 @@
+        required=False,
+    )
+
+    parser.add_argument(
+        "-testPercent",
+        type=float,
+        default=0.1,
+        help="Percentage of the labeled data used for testing",
+        required=False,
+    )
+
+    parser.add_argument(
+        "-validationPercent",
+        type=float,
+        default=0.05,
+        help="Percentage of the labeled data used for testing",
+        required=False
+    )
+
+    parser.add_argument(
+        "-numClasses",
+        type=float,
+        default=150,
+        help="Number of unique classes (venue labels)",
+        required=False,
+    )
+
+    # --- Multi-GPU options
+    parser.add_argument(
+        "-mg",
+        type=bool,
+        default=False,
+        help="Run Multi-GPU. This will create a local DASK cluster."
+             " Single GPU is limited to Scale 26",
+        required=False,
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+# ------------------------------------------------
+
+
+def main() -> None:
+
+    args = parse_args()
+
+    # --- Step 0: Setup/Init ---
+    if args.mg:
+        client, cluster = setup_mg()
+    else:
+        setup_sg()
+
+    print(f"Running in MG mode is {args.mg}")
+
+    # Create directories
+    setup_directories(args.outdir)
+
+    # Step 1: Create the Papers Cite graph
+    num_paper_cites_paper_edges, last_paper_id = create_paper_cites_data(args)
+    num_papers = last_paper_id + 1
+
+    # --- Step 2: Papers info  (labels and features) ---
+    train_ix, test_ix, val_ix = create_papers_data(args, num_papers)
+
+    # --- Step 3: Paper Labels (venue id) ---
+    create_paper_labels(args, train_ix, test_ix, val_ix)
+
+    # --- Step 4:Author to Papers ---
+    num_author_writes_paper_edges, num_auth = (
+        create_author_papers_data(args, (last_paper_id + 1), num_papers)
+    )
+
+    # --- Step 5:Author to Institutions ---
+    next_id = num_papers + num_auth + 1
+    num_author_affil_institution_edges, num_inst = (
+        create_author_institute_data(args, (next_id), num_auth)
+    )
+
+    with open(os.path.join(args.outdir, 'meta.json'), 'w') as out_json_file:
+        json.dump(
+            {
+                'paper': [
+                    0,
+                    int(last_paper_id)
+                ],
+                'author': [
+                    int(last_paper_id) + 1,
+                    int(last_paper_id + num_auth)
+                ],
+                'institution': [
+                    int(last_paper_id + num_auth) + 1,
+                    int(last_paper_id + num_auth + num_inst) - 1
+                ],
+                'author__writes__paper': num_author_writes_paper_edges,
+                'author__affiliated_with__institution':
+                    num_author_affil_institution_edges,
+                'paper__cites__paper': num_paper_cites_paper_edges,
+                'train': train_ix.tolist(),
+                'test': test_ix.tolist(),
+                'val': val_ix.tolist(),
```

Comment:
We are writing these to disk as numpy arrays . I dont think we should write these in json as this wont scale and makes parsing a dict for large graphs prohibitive.

Reply:
We currently are not writing these to disk.  I would have to change that.

Reply:
The numpy files contain the train/test/val labels but not the indices.  But I can modify those files.

Reply:
> 1. We currently are not writing these to disk. I would have to change that.

I think we are , see lines:  https://github.com/rapidsai/cugraph/blob/6902ffea35b56b33b96bd3b12a1c27b6c6b23ebb/benchmarks/shared/python/data_generators/gnn_cite_graph_gen.py#L792-L795


> The numpy files contain the train/test/val labels but not the indices. But I can modify those files.

Yes please. I wrote my sampler on the assumption that the train/test/ival  numpy files contains indices and not labels . I will modify it but we should have 2 files, one with **labels** and one with corresponding  **indices**  . 

We need to consistent mapping.

Reply:
I'm almost done fixing it, just give me a minute.  Why don't you want a single parquet with values and indices together?

Reply:
That also works.

Reply:
ok, great

Reply:
give me about a minute

Reply:
ok, should be good to go noe

Reply:
I see train_ix.tolist() , am i missing something ?